pkg/sidecar/sidecar.go

```diff
@@ -235,23 +170,22 @@ func (s *Sidecar) checkProcessExit() {
 // the Workload API, and calls writeCerts and writeKey to write to disk
 // the svid, key and bundle of certificates.
 // It is possible to change output setting `addIntermediatesToBundle` as true.
-func (s *Sidecar) dumpBundles(svidResponse *proto.X509SVIDResponse) error {
+func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	// There may be more than one certificate, but we are interested in the first one only
-	svid := svidResponse.Svids[0]
+	svid := svidResponse.DefaultSVID()
 
 	svidFile := path.Join(s.config.CertDir, s.config.SvidFileName)
 	svidKeyFile := path.Join(s.config.CertDir, s.config.SvidKeyFileName)
 	svidBundleFile := path.Join(s.config.CertDir, s.config.SvidBundleFileName)
 
-	certs, err := x509.ParseCertificates(svid.X509Svid)
-	if err != nil {
-		return err
-	}
-
-	bundles, err := x509.ParseCertificates(svid.Bundle)
-	if err != nil {
-		return err
+	certs := svid.Certificates
+	bundleSet, found := svidResponse.Bundles.Get(svid.ID.TrustDomain())
+	if !found {
+		return fmt.Errorf("no bundles found for %s trust domain", svid.ID.TrustDomain().String())
 	}
+	bundles := bundleSet.X509Authorities()
+	privateKey := svid.PrivateKey.(crypto.PrivateKey)
```

Comment:
x509.MarshalPKCS8PrivateKey takes `interface {}` so this type assertion isn't necessary.